lib/galaxy/tools/__init__.py

```diff
@@ -1860,8 +1860,8 @@ def populate_model( inputs, state_inputs, group_inputs, other_values=None ):
 
         # create tool model
         tool_model = self.to_dict( request_context )
-        tool_model[ 'inputs' ] = []
-        populate_model( self.inputs, state_inputs, tool_model[ 'inputs' ] )
+        tool_model[ 'inputs' ] = {}
+        self.populate_model( request_context, self.inputs, state_inputs, tool_model[ 'inputs' ] )
```

Comment:
Yes, this should be fixed. I'll be having a look at that today and tomorrow. I remember the `populate_model` method being an annoying source of frequent merge conflicts, so it's no surprise that one of those merges went the wrong way and broke things.